### PR TITLE
code refactoring:

### PR DESCRIPTION
**Pull request by rogernkosi**
clean white space
extract location changed logic to a separate method on the presenter

### CODE REVIEW

File: app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt [hunk 1]
```diff
@@ -55,48 +48,35 @@ class NearbyAirportsActivity : BaseFragmentActivity<NearbyPresenter>(), NearbyVi
...
-        this.googleMap.setMinZoomPreference(2f)
-        // Set a listener for marker click.
-        this.googleMap.setOnMarkerClickListener(this)
     }
 
     override fun instantiatePresenter(): NearbyPresenter {
         return NearbyPresenter(this)
     }
 
     override fun getAirports(airport: List<Airport>) {
```
Comment (N2Android) [inline app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt]: Rename the ```airport``` variable to something like ```airportList```.

File: app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt [hunk 2]
```diff
@@ -55,48 +48,35 @@ class NearbyAirportsActivity : BaseFragmentActivity<NearbyPresenter>(), NearbyVi
...
-
-        Log.e("airport", "air.nameAirpor")
-
+        Log.e("airport", "air.nameAirport")
```
Comment (N2Android) [inline app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt]: I feel like this log is redundant? Unless you adding it as the start debugging log? - We should later once the functionality works remove all these logs. Or we can use a lib like Timber to allow configuring dev logs.

File: app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt [hunk 3]
```diff
@@ -55,48 +48,35 @@ class NearbyAirportsActivity : BaseFragmentActivity<NearbyPresenter>(), NearbyVi
...
     override fun onMapReady(googleMap: GoogleMap) {
-
         if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
             ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION), locationRequestCode)
-            return
+        } else {
+            this.googleMap = googleMap
+            this.googleMap.isMyLocationEnabled = true
+            this.googleMap.uiSettings.isMyLocationButtonEnabled = true
+            this.googleMap.setMinZoomPreference(2f)
```
Comment (N2Android) [inline app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt]: Remember to extract the ```2f``` to a variable explaining the config.

File: app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt (hunk 2 above)
Comment (rogernkosi) [reply app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt]: yes this log as the start of debugging, but it is not needed anymore

File: app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt [hunk 4]
```diff
@@ -123,110 +102,72 @@ class NearbyAirportsActivity : BaseFragmentActivity<NearbyPresenter>(), NearbyVi
...
-        val bitmap = Bitmap.createBitmap(drawable.intrinsicWidth, drawable.intrinsicHeight, Bitmap.Config.ARGB_8888)
-        canvas.setBitmap(bitmap)
-        drawable.setBounds(0, 0, drawable.intrinsicWidth, drawable.intrinsicHeight)
-        drawable.draw(canvas)
-        return BitmapDescriptorFactory.fromBitmap(bitmap)
-    }
-
-
     private fun setUpLocationProvider() {
         // Get the location manager
```
Comment (N2Android) [inline app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyAirportsActivity.kt]: You can remove this comment as well - is self explanatory.

File: app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyPresenter.kt [hunk 5]
```diff
@@ -29,22 +36,42 @@ class NearbyPresenter(nearbyView: NearbyView) : BasePresenter<NearbyView>(nearby
             .subscribe({ airportList -> view.getAirports(airportList) },
                 { s -> view.showError(s) }
             )
-
     }
 
     /**
      * This method may be called when the presenter view is created
      */
     override fun onViewCreated() {
```
Comment (N2Android) [inline app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyPresenter.kt]: If this method and interface is unused please remove - only add it again if specific use case.

File: app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyPresenter.kt [hunk 6]
```diff
@@ -29,22 +36,42 @@ class NearbyPresenter(nearbyView: NearbyView) : BasePresenter<NearbyView>(nearby
...
-
     }
 
     /**
```
Comment (N2Android) [inline app/src/main/java/com/example/flightstimetable/ui/nearby/NearbyPresenter.kt]: Remove comment

**Comment by N2Android** [comment]
Except for the last comments the rest looks fine. Please make those changes and then you can merge it in.

**Pull request merged**